### PR TITLE
SOC-1984 | Unify color response from MercuryAPI to mercury

### PR DESCRIPTION
**Pull request by Bloodmoon24**
https://wikia-inc.atlassian.net/browse/SOC-1984

@kvas-damian  @bkoval  please CR

### CODE REVIEW

**Comment by kvas-damian** [comment]
LGTM :shipit:

File: extensions/wikia/SASS/SassUtil.php [hunk 1]
```diff
@@ -453,4 +455,19 @@ public static function colorNameToHex($colorName) {
...
 
+	/**
+	 * @desc Converts theme designer color names to hex
+	 *
+	 * @param $themeSettings
+	 * @return mixed
+	 */
+	public static function normalizeThemeColors($themeSettings) {
+		foreach (self::THEME_DESIGNER_COLOR_KEYS as $key) {
+			if( !preg_match(self::HEX_REG_EXP, $themeSettings[$key])) {
```
Comment (bkoval) [inline extensions/wikia/SASS/SassUtil.php]: Don't we do space after if in PHP? :)

File: extensions/wikia/SASS/SassUtil.php [hunk 2]
```diff
@@ -453,4 +455,19 @@ public static function colorNameToHex($colorName) {
...
 	}
 
+	/**
+	 * @desc Converts theme designer color names to hex
+	 *
+	 * @param $themeSettings
+	 * @return mixed
+	 */
+	public static function normalizeThemeColors($themeSettings) {
+		foreach (self::THEME_DESIGNER_COLOR_KEYS as $key) {
```
Comment (bkoval) [inline extensions/wikia/SASS/SassUtil.php]: Spaces inside brackets

File: extensions/wikia/SASS/SassUtil.php [hunk 3]
```diff
@@ -453,4 +455,19 @@ public static function colorNameToHex($colorName) {
 		}
 	}
 
+	/**
+	 * @desc Converts theme designer color names to hex
+	 *
+	 * @param $themeSettings
+	 * @return mixed
+	 */
+	public static function normalizeThemeColors($themeSettings) {
```
Comment (bkoval) [inline extensions/wikia/SASS/SassUtil.php]: Spaces inside brackets.

File: extensions/wikia/SASS/SassUtil.php [hunk 4]
```diff
@@ -453,4 +455,19 @@ public static function colorNameToHex($colorName) {
...
+		foreach (self::THEME_DESIGNER_COLOR_KEYS as $key) {
+			if( !preg_match(self::HEX_REG_EXP, $themeSettings[$key])) {
+				$themeSettings[$key] = self::colorNameToHex($themeSettings[$key]);
```
Comment (bkoval) [inline extensions/wikia/SASS/SassUtil.php]: I'm not sure how about invoking function, but there might be a need for spacing in the round brackets as well :)

File: extensions/wikia/MercuryApi/models/MercuryApi.class.php [hunk 5]
```diff
@@ -134,7 +134,7 @@ public function getWikiVariables() {
 			'namespaces' => $wgContLang->getNamespaces(),
 			'siteMessage' => $this->getSiteMessage(),
 			'siteName' => $wgSitename,
-			'theme' => SassUtil::getOasisSettings(),
+			'theme' => SassUtil::normalizeThemeColors(SassUtil::getOasisSettings()),
```
Comment (bkoval) [inline extensions/wikia/MercuryApi/models/MercuryApi.class.php]: Spaces inside round brackets :)

**Comment by bkoval** [comment]
LGTM

**Pull request merged**